From 35ccc8e0d04967b923f59ae63ef4b545fd94e748 Mon Sep 17 00:00:00 2001
From: Wyon Bi <bivvy.bi@rock-chips.com>
Date: Wed, 7 Nov 2018 19:59:16 +0800
Subject: [PATCH] drm/rockchip: dsi: ignore return value of drm_panel_attach()

drm_panel_attach() return EBUSY if panel->connector already set,
which indicate that panel and connector are already bound.
Because the .bind() function might call multiple times, so it is
unnecessary to check its return value.

Change-Id: I45e82983dc177825561ff4796e585dafe0f7c201
Signed-off-by: Wyon Bi <bivvy.bi@rock-chips.com>
---
 drivers/gpu/drm/rockchip/dw-mipi-dsi.c | 9 +--------
 1 file changed, 1 insertion(+), 8 deletions(-)

diff --git a/drivers/gpu/drm/rockchip/dw-mipi-dsi.c b/drivers/gpu/drm/rockchip/dw-mipi-dsi.c
index 4518d46d040d..2e3a379905c5 100644
--- a/drivers/gpu/drm/rockchip/dw-mipi-dsi.c
+++ b/drivers/gpu/drm/rockchip/dw-mipi-dsi.c
@@ -1517,18 +1517,11 @@ static int dw_mipi_dsi_register(struct drm_device *drm,
 		drm_connector_helper_add(connector,
 					 &dw_mipi_dsi_connector_helper_funcs);
 		drm_mode_connector_attach_encoder(connector, encoder);
-
-		ret = drm_panel_attach(dsi->panel, &dsi->connector);
-		if (ret) {
-			dev_err(dev, "Failed to attach panel: %d\n", ret);
-			goto connector_cleanup;
-		}
+		drm_panel_attach(dsi->panel, connector);
 	}
 
 	return 0;
 
-connector_cleanup:
-	drm_connector_cleanup(connector);
 encoder_cleanup:
 	drm_encoder_cleanup(encoder);
 	return ret;
-- 
2.35.3

